### DISCUSSION TITLE
Swapped Face is blurry in the Result

### DISCUSSION THREAD

**Discussion by lost-in-emotions** [glucauze/sd-webui-faceswaplab#132] [Q&A]
hey there, i can see the source face in my swapped image, so far so good. but everything i do it gets kinda blurry... or more blurry as the rest of the generated picture (txt2img). i used everywhere 4x-ultrasharp upscaler, but that doesnt help. the source face that i use has 3000x3000 px and 3MB file...

whats the cause that the swapped face (of my source) is more blurry as the rest in the target result? 

thanks for suggestions...

**Comment by Rifz42** [comment]
I think it only uses 128x128 images to start with..  see this.
https://github.com/glucauze/sd-webui-faceswaplab/discussions/135#discussioncomment-7256834

**Comment by coffeedrinker0725** [comment]
There are Advanced Options : Under Post-Processing & Advanced Mask Options, click on "codeformer" or "gfpgan", then select upscaler: ESRGAN_4x,
furthermore, click on "sharpen face", then click generate.
